samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -123,31 +189,109 @@ bool BridgeStorageManager::RemoveBridgedDeviceType(uint8_t bridgedDeviceIndex)
 	return Nrf::GetPersistentStorage().NonSecureRemove(&id);
 }
 
-bool BridgeStorageManager::StoreBridgedDevice(const MatterBridgedDevice *device, uint8_t index)
+bool BridgeStorageManager::LoadBridgedDevice(BridgedDevice &device, uint8_t index)
 {
-	if (!device) {
+	Nrf::PersistentStorageNode id = CreateIndexNode(index, &mBridgedDevice);
+	size_t readSize = 0;
+	uint8_t buffer[sizeof(BridgedDevice) + kMaxUserDataSize];
+	uint16_t counter = 0;
+	const uint8_t mandatoryItemsSize = sizeof(device.mEndpointId) + sizeof(device.mDeviceType) + sizeof(device.mNodeLabelLength) + sizeof(device.mUserDataSize);
+
+	/* Validate that read size is big enough to include mandatory data. */
+	if (readSize < mandatoryItemsSize) {
 		return false;
 	}
 
-	if (!StoreBridgedDeviceEndpointId(device->GetEndpointId(), index)) {
+	if (!Nrf::GetPersistentStorage().NonSecureLoad(&id, buffer, sizeof(BridgedDevice) + kMaxUserDataSize, readSize)) {
 		return false;
 	}
 
-	if (!StoreBridgedDeviceNodeLabel(device->GetNodeLabel(), strlen(device->GetNodeLabel()), index)) {
+	/* Deserialize data and copy it from buffer into structure's fields. */
+	memcpy(&device.mEndpointId, buffer, sizeof(device.mEndpointId));
+	counter += sizeof(device.mEndpointId);
+	memcpy(&device.mDeviceType, buffer + counter, sizeof(device.mDeviceType));
+	counter += sizeof(device.mDeviceType);
+	memcpy(&device.mNodeLabelLength, buffer + counter, sizeof(device.mNodeLabelLength));
+	counter += sizeof(device.mNodeLabelLength);
+
+	/* Validate that read size is big enough to include all expected data. */
+	if (readSize < mandatoryItemsSize + device.mNodeLabelLength) {
+		return false;
+	}
+
+	memcpy(device.mNodeLabel, buffer + counter, device.mNodeLabelLength);
+	counter += device.mNodeLabelLength;
+
+	/* Check if user prepared a buffer for reading user data. It can be nullptr if not needed. */
+	if (!device.mUserData) {
+		device.mUserDataSize = 0;
+		return true;
+	}
+
+	uint8_t inUserDataSize = device.mUserDataSize;
+	/* Validate if user buffer size is big enough to fit the stored user data size. */
+	if (device.mUserDataSize + counter + sizeof(device.mUserDataSize) < readSize ) {
```

Comment:
```suggestion
	if (readSize < counter + sizeof(device.mUserDataSize)) {
```
Given that you check the user buffer further (in line 241) it's enough to check that you can read the user data length.